src/Mod/PartDesign/Gui/TaskChamferParameters.cpp

```diff
@@ -194,6 +198,31 @@ void TaskChamferParameters::clearButtons(const selectionModes notThis)
     DressUpView->highlightReferences(false);
 }
 
+// By adding the edges to the Gui.Selection in python, then onSelectionChanged() gets called
+// where the edges get added to the list in the task dialog and highlighted in the 3d view
+
+void TaskChamferParameters::onSelectAllEdges(void)
+{
+    PartDesign::Chamfer* pcChamfer = static_cast<PartDesign::Chamfer*>(DressUpView->getObject());
+    if (pcChamfer) {
+        App::DocumentObject* base = pcChamfer->Base.getValue();
+        if (base) {
+            std::string baseName = base->getNameInDocument();
+            std::ostringstream command;
+            command << "from PySide import QtGui,QtCore\n";
+            command << "QtGui.QApplication.setOverrideCursor(QtCore.Qt.WaitCursor)\n";
+            command << "for _ii in range(0, len(App.ActiveDocument.getObject(\'" << baseName << "\').Shape.Edges)):\n";
+            command << "    Gui.Selection.addSelection(App.ActiveDocument.getObject(\'" << baseName << "\'), \'Edge\'+str(_ii+1))\n";
+            command << "    if _ii % 20:\n";

```

Comment:
isn't it extremely slow to add each edge separately? Why not directly updating the QListWidget without the detour of the selection?